pkg/controllers/nodeclaim/disruption/drift_test.go

```diff
@@ -112,8 +112,12 @@ var _ = Describe("Drift", func() {
 	It("should detect stale instance type drift if the instance type offerings aren't compatible with the nodeclaim", func() {
 		cp.InstanceTypes = lo.Map(cp.InstanceTypes, func(it *cloudprovider.InstanceType, _ int) *cloudprovider.InstanceType {
 			if it.Name == nodeClaim.Labels[corev1.LabelInstanceTypeStable] {
-				newLabels := lo.Keys(nodeClaim.Labels)
-				it.Requirements = scheduling.NewLabelRequirements(map[string]string{newLabels[0]: test.RandomName()})
+				for i := range it.Offerings {
```

Comment:
Oh! This actually isn't good at all -- looks like the default state that we set the generic cloudprovider to is `"drifted"` after a `Reset()` which means that regardless of whether the test _should_ actually pass for the drifted condition that it's checking, it's still going to return drifted at all times in the cloudprovider reason -- probably best to check the reason for "Drifted" in this test as well (in the status condition) so that we avoid cases like this

Reply:
We should absolutely update that cloudprovider.Reset() function though to not return "drifted" by default

Reply:
it sounds like a former work required. feel free to amend or just let me know after that done.

i will take another one from open issues.

Reply:
Just pushed the change -- one comment, then I think this looks like it's good to go